File: 30-days-of-code/day-25---running-time-and-complexity/solution.py
import math


def is_prime(n):
    if n == 1:
        return 'Not prime'
    if n == 2:
        return 'Prime'
    if n % 2 == 0:
        return 'Not prime'

    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return 'Not prime'
    return 'Prime'

length = int(input())

for j in range(length):
    num = int(input())
    print(is_prime(num))

File: 30-days-of-code/day-25---running-time-and-complexity/solution.js
function processData(input) {
    const data = input.split('\n');
    const length = parseInt(data[0]);

    for (let i=1; i<=length; i++) {
        const num = parseInt(data[i]);
        console.log(isPrime(num));
    }
}

function isPrime(n) {
    if (n == 1) return 'Not prime';
    if (n == 2) return 'Prime';
    if (n % 2 == 0) return 'Not prime';

    for (let i=3; i <= Math.sqrt(n); i+=2) {
        if (n % i == 0) return 'Not prime';
    }
    return 'Prime';
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});
